realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1092,6 +1092,20 @@ public boolean isValid() {
         return this;
     }
 
+    /**
+     * Condition that finds a distinct set of objects of a specific class. As a Realm is unordered,
+     * it is undefined which objects are returned in case of multiple occurrences.
+     *
+     * @param fieldName the field name.
+     * @return the query object.
+     * @throws IllegalArgumentException if a field name does not exist or the field is not indexed.
+     */
+    public RealmQuery<E> distinct(String fieldName) {
+        long columnIndices[] = schema.getColumnIndices(fieldName, RealmFieldType.STRING);
+        this.query.distinct(columnIndices);
+        return this;
```

Comment:
That's what I was wondering as well. But for example, can you go with something like `realm.where(Foo.class).lessThanOrEqualTo("something", new Date()).distinct("something")`?

The semantic of how and where to put `distinct` confused me a bit.